### DISCUSSION TITLE
Loop In

### DISCUSSION THREAD

**Discussion by bilthon** [Ride-The-Lightning/RTL#1267] [Troubleshooting Questions] [answered]
I've used the loop-out feature a few times now. But it's not that clear how the loop-in works. I'm given an HTLC address which I can only assume must be funded by me at some point. But it's also not clear to me when will the HTLC expire, and what happens if I happen to fund it after it expires. Being an HTLC I suppose I'd be able to sweep it back somehow, but does my wallet need to understand this somehow?

I was going to fund it with electrum by bundling it with another transaction I was going to make. But I realized it quickly went from a "HTLC Published" state to "Failed" so I backed off. I also could not find any tutorial or resource online.

**Comment by saubyk** [comment]
Hi, @bilthon have you looked at the loop section on the lightning labs website?
https://lightning.engineering/loop/
Also there are a bunch of blogs available on the lightning labs site on loop:
https://lightning.engineering/posts/2019-06-25-loop-in/
https://lightning.engineering/posts/2019-03-20-loop/
https://lightning.engineering/posts/2020-02-05-loop-beta/

**Reply by bilthon** [reply]
Well yes, I've been over those and they of course don't answer my questions. I think it's very lazy to point someone with very specific questions to a generic documentation. If you don't want people using this service you could have saved the trouble and avoid integrating it in the first place.

**Reply by saubyk** [reply, marked as answer]
- you started by saying that you're not clear how loop-in works, without providing any background on what documentation you've gone through, so I linked a few docs based on my assumptions
- Loop-in currently works on RTL, if you have sufficient funds on your LND on-chain wallet. It doesn't work the way **you** specifically are trying to make it work and that's not a problem we can solve for you
- If you don't like the way service is integrated with RTL, feel free not to use it. No need to lecture us here on what we should or shouldn't do

**Comment by bilthon** [comment]
Sorry I should have been more explicit stating that I'm not clear how loop-in works in the context of RTL.

But by the context provided by the rest of the message one could guess that I was not completely unaware of the operating principle behind a loop-in. In fact I understood just enough to shoot me in the foot there. Had I funded it via an external wallet my money would probably have been locked in there. Luckily I backed off and went to the CLI tool instead.

I understand why showing the HTLC address is important, but a warning for users **NOT** to fund it externally should be considered.
